src/django/api/views.py

```diff
@@ -1200,6 +1223,78 @@ def update_location(self, request, pk=None):
         facility_data = FacilityDetailsSerializer(facility).data
         return Response(facility_data)
 
+    @waffle_switch('facility_history')
+    @action(detail=True, methods=['GET'],
+            permission_classes=(IsRegisteredAndConfirmed,),
+            url_path='history')
+    def get_facility_history(self, request, pk=None):
+        """
+        Returns the history of changes to a facility as a list of dictionaries
+        describing the changes.
+
+        ### Sample Response
+            [
+                {
+                    "updated_at": "2019-09-12T02:43:19Z",
+                    "action": "DELETE",
+                    "detail": "Deleted facility"
+                },
+                {
+                    "updated_at": "2019-09-05T13:15:30Z",
+                    "action": "UPDATE",
+                    "changes": {
+                        "location": {
+                            "old": {
+                                "type": "Point",
+                                "coordinates": [125.6, 10.1]
+                            },
+                            "new": {
+                                "type": "Point",
+                                "coordinates": [125.62, 10.14]
+                            }
+                        }
+                    },
+                    "detail": "FacilityLocation was changed"
+                },
+                {
+                    "updated_at": "2019-09-02T21:04:30Z",
+                    "action": "MERGE",
+                    "detail": "Merged with US2019123AG4RD"
+                },
+                {
+                    "updated_at": "2019-09-01T21:04:30Z",
+                    "action": "CREATE",
+                    "detail": "Facility was created"
+                }
+            ]
+        """
+        try:
+            # check to ensure that the endpoint sends back a 404
+            # for facilities which have never existed
+            # Facility.history.get(id=pk)
+
+            facility_history = create_facility_history_list(
+                Facility.history.filter(id=pk),
+                pk,
+            )
+
+            if len(facility_history) == 0:
+                raise NotFound()
+
+            return Response(facility_history)
+        except HistoricalFacility.DoesNotExist:
+            raise NotFound()
+        except:  # noqa
+            try:
+                Facility.history.get(id=pk)
+            except HistoricalFacility.DoesNotExist:
+                raise NotFound()
```

Comment:
I adjusted this in bb223f25d12c8b9f41eb4f8ae0a90f4ddfddee37

Now it:

- checks for historical records when filtering id=pk prior to the try block
- immediately raises NotFound if the record count is 0 (since that is the sign that the facility does not exist)
- does the history response assembly inside the try block
- allows any exceptions encountered there to bubble up to the last except clause